ldk/javascript/src/whisper/react/whisper-render-instance.ts

```diff
@@ -0,0 +1,58 @@
+import { NewWhisper, UpdateWhisper, Whisper } from '../types';
+import { create } from '../index';
+
+export interface WhisperRenderingInterface {
+  createOrUpdateWhisper(whisperData: NewWhisper | UpdateWhisper): void;
+  closeWhisper(): void;
+  setOnClose(func: () => Promise<void>): void;
+}
+
+enum RenderInstanceStatus {
+  NotCreated,
+  Created,
+  Closed,
+}
+
+/**
+ * @internal
+ */
+export class WhisperRenderInstance implements WhisperRenderingInterface {
+  private status: RenderInstanceStatus = RenderInstanceStatus.NotCreated;
+
+  private whisper: Whisper | undefined;
+
+  private onCloseHandler: undefined | (() => Promise<void>);
+
+  async createOrUpdateWhisper(whisperData: NewWhisper | UpdateWhisper): Promise<void> {
+    if (this.status === RenderInstanceStatus.Closed) {
+      return;
+    }
+    if (this.whisper == null) {
+      await this.createWhisper(whisperData as NewWhisper);
+    } else {
+      if (this.status !== RenderInstanceStatus.Created) {
```

Comment:
Redundant if layer? else if could be used

Reply:
Yep! Good catch :)